sql/2021/pwa/sw_adoption_over_time.sql

```diff
@@ -0,0 +1,15 @@
+#standardSQL
+# SW adoption over time - based on 2019/11_01b.sql
+SELECT
+  yyyymmdd AS date,
+  client,
+  num_urls AS freq,
+  total_urls AS total,
+  pct_urls AS pct
+FROM
+  `httparchive.blink_features.usage`
+WHERE
+  feature = 'ServiceWorkerControlledPage'

```

Comment:
This is one of the mos interesting queries. Taking a look at [the chart](https://www.chromestatus.com/metrics/feature/timeline/popularity/990), we might be able to reach to 20% of pages controlled by a service worker by the time the Almanac launches, which is a great stat. A couple of thoughts:

- I recall you've mentioned that this year we would be able to see if the top sites use service workers, in order to support our hypothesis that, even when 1% of URLs have service workers, 20% of traffic is controlled, because top sites implemented it. Is that correct?
- We asked internally but we were not able to explain why from [Nov. 2019 and May 2020](https://www.chromestatus.com/metrics/feature/timeline/popularity/990) there's a pronounced drop in usage, that is restored afterwards. Maybe we can leave a note indicating that we don't have a reason to explain this?

Reply:
> I recall you've mentioned that this year we would be able to see if the top sites use service workers, in order to support our hypothesis that, even when 1% of URLs have service workers, 20% of traffic is controlled, because top sites implemented it. Is that correct?

I've added a query for this for top 1,000 sites in https://github.com/HTTPArchive/almanac.httparchive.org/pull/2272/commits/e519676de7e4c076341c953a56dfaab6c4339454. Note this is only available from May this year.

All sites:

date | client | freq | total | pct
-- | -- | -- | -- | --
20210601 | desktop | 74844 | 6435567 | 1.16%
20210601 | mobile | 84998 | 7452047 | 1.14%
20210501 | desktop | 70558 | 6389672 | 1.10%
20210501 | mobile | 79878 | 7364482 | 1.08%

Top 1000 sites (note not actually 1000 since we can't reach some sites):

date | client | freq | total | pct
-- | -- | -- | -- | --
20210601 | desktop | 60 | 803 | 7.47%
20210601 | mobile | 72 | 942 | 7.64%
20210501 | desktop | 60 | 796 | 7.54%
20210501 | mobile | 73 | 938 | 7.78%

An interesting difference I think you'll agree!

> We asked internally but we were not able to explain why from Nov. 2019 and May 2020 there's a pronounced drop in usage, that is restored afterwards. Maybe we can leave a note indicating that we don't have a reason to explain this?

It's interesting that we only see that drop in the Chrome usage, and not in the [HTTP Archive crawl](https://httparchive.org/reports/progressive-web-apps#swControlledPages). This suggests either Chrome stopped logging this correctly, or a big player (e.g. Facebook) stopped using Service Workers for a period. Unfortunately as we don't have access to the underlying data of Chrome's usage we can't take this any further.

Last year we didn't use Chrome's usage graph, and use the crawl graph, and only refered to Chrome's usage: "While that usage may seem low, it is important that we realize that other measurements equate that to 16.6% of the [web traffic](https://www.chromestatus.com/metrics/feature/timeline/popularity/990)–the difference being due to high traffic websites tending to use service workers more."

Probably best to do the same thing year to avoid confusing readers with this anomaly.

> How far in the past can we go with this report? It would be interesting to see it as a timeseries, starting when service workers were launched (around 2015), to our days.

We only have `blink_usage` stats back to January 2017 but even then it's down to a couple of hundred sites so doubt it's useful before that.

Reply:
In fact here's a better graph looking at all the ranking categories for June 2021:

![Service Worker Controlled Pages by Ranking](https://user-images.githubusercontent.com/10931297/126149438-807000e6-f901-4c7c-a180-1280ac4d2107.png)

And here's a timeline of the mobile results (though less interesting as only have two months worth of data at present):

![SW Controlled Pages by Ranking over Time](https://user-images.githubusercontent.com/10931297/126149749-ef2746a0-dcc8-46ca-9f08-9c04e71d4924.png)

Reply:
All these comments are great, thanks for working on this @tunetheweb!

I like the chart showing how the SW usage decreases as one broadens the scope of the search beyond the top sites. I think we can use it in the SW section to explain why the number of URLs controlled by SW differs from the actual percentage of page loads.

I think we are good in terms of this section and queries 👍

Reply:
@tunetheweb: one more question regarding this: do you have the query to get the list of top 100 / 1000 sites?
I'd like to go over them "manually" to see if there's any interesting pattern to extract.

Reply:
Saved here (along with the query needed to get it):
https://docs.google.com/spreadsheets/d/10KT5CxH6VKRktg90MwfzXuSXn67pxUuNiIxR94B5FSg/edit?usp=sharing

Reply:
Thanks for sharing the list @tunetheweb! I went over it, and taking a look at the origins listed there, it makes sense that a 1% of the URLs controlled by service workers, represent almost 20% of the web traffic.

As you know, the limitation with that list is that it leaves out potentially high traffic sites that only register the service worker after the user signs in. This is the case of Facebook, for example.

In that case, the URL won't show up, but the traffic will still be counted for the 20% of traffic controlled by service workers.

In general, this is a very good approximation, though.